ui/app/helpers/confirm-transaction/util.js

```diff
@@ -137,3 +138,11 @@ export function convertTokenToFiat ({
 export function hasUnconfirmedTransactions (state) {
   return unconfirmedTransactionsCountSelector(state) > 0
 }
+
+export function roundExponential (value) {
+  const PRECISION = 4
+  const valueBN = new BigNumber(value)
+
+  // In JS, numbers with exponentials greater than 20 get displayed as an exponential.
+  return valueBN.e > 20 ? Number(valueBN.toPrecision(PRECISION)) : value
```

Comment:
If `valueBN.e` is less than twenty, but great than (for instance) 16, will we still run into overflow issues? Has rounding been properly handled elsewhere for all cases where `valueBN.e < 20`?

Reply:
If `valueBN.e < 20`, the number isn't represented as a decimal so I don't think rounding is appropriate, since you have the exact number of tokens being sent. The purpose of this check is to round only when the number is being represented as an exponential.

Reply:
So what do you do with the overflow?